core/src/main/java/org/elasticsearch/percolator/MultiDocumentPercolatorIndex.java

```diff
@@ -71,7 +72,16 @@ public void prepare(PercolateContext context, ParsedDocument parsedDocument) {
         try {
             MultiReader mReader = new MultiReader(memoryIndices, true);
             LeafReader slowReader = SlowCompositeReaderWrapper.wrap(mReader);
-            final IndexSearcher slowSearcher = new IndexSearcher(slowReader);
+            final IndexSearcher slowSearcher = new IndexSearcher(slowReader) {
+
+                @Override
+                public void search(Query query, Collector collector) throws IOException {
+                    BooleanQuery.Builder bq = new BooleanQuery.Builder();
+                    bq.add(query, BooleanClause.Occur.MUST);
+                    bq.add(Queries.newNestedFilter(), BooleanClause.Occur.MUST_NOT);
+                    super.search(bq.build(), collector);
```

Comment:
The call to preCreate skews indexing stats. You must call postCreate to decrement the counters increased in preCreate. Problem is that I don't think that indexing operations from the translog should be counted in the stats api. these would ideally report only use actions. Thinking about how to work around this, I wonder if we should have a callback for extra parsers - i.e., things that extract more info out of source and index it in lucene. It seems that this is what we're doing here?